### PR TITLE
Allow completed date filter to be added for task list

### PR DESCRIPTION
**Pull request by amohammedalfresco**
[Activiti/issues#3447](https://github.com/Activiti/Activiti/issues/3447)

### CODE REVIEW

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java [hunk 1]
```diff
@@ -1661,4 +1661,60 @@ public void should_getTask_when_queryFilteredByPriority() {
 
     }
 
+    @Test
+    public void shouldGetProcessInstancesFilteredByCompletedDate() {
```
Comment (nithinnambiar) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: Test can be renamed as it is filtering Tasks and not ProcessInstances

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java [hunk 2]
```diff
@@ -1661,4 +1661,60 @@ public void should_getTask_when_queryFilteredByPriority() {
...
+        Date completedDateToday = new Date();
+        Date completedDateTwoDaysAgo = new Date();
+        Date completedDateFiveDaysAfter = new Date();
+
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        Date now = cal.getTime();
+
+        //Start a process and set it's completed date as current date
```
Comment (nithinnambiar) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: //start a Task

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java [hunk 3]
```diff
@@ -1661,4 +1661,60 @@ public void should_getTask_when_queryFilteredByPriority() {
...
+        completedDateToday.setTime(now.getTime());
+        Task task1 = taskEventContainedBuilder.aCompletedTaskWithCompletionDate("Task1", runningProcessInstance, completedDateToday);
+
+        //Start a process and set it's completed date as current date minus two days
```
Comment (nithinnambiar) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: //start a Task

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java [hunk 4]
```diff
@@ -1661,4 +1661,60 @@ public void should_getTask_when_queryFilteredByPriority() {
...
+        completedDateTwoDaysAgo.setTime(now.getTime() - Duration.ofDays(2).toMillis());
+        taskEventContainedBuilder.aCompletedTaskWithCompletionDate("Task2", runningProcessInstance, completedDateTwoDaysAgo);
+
+        //Start a process and set it's completed date as current date plus five days
```
Comment (nithinnambiar) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: //start a Task

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java (hunk 1 above)
Comment (amohammedalfresco) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: Good catch, thanks.  I will fix it.

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java (hunk 2 above)
Comment (amohammedalfresco) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: Fixed.

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java (hunk 3 above)
Comment (amohammedalfresco) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: Fixed.

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java (hunk 4 above)
Comment (amohammedalfresco) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java]: Fixed.

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**